pkg/receiver/gitproviderreceiver/internal/scraper/githubscraper/github_scraper.go

```diff
@@ -62,58 +62,88 @@ func newGitHubScraper(
 	}
 }
 
-func (ghs *githubScraper) getPullRequests(
+func getTotalPrPages(
+	ghs *githubScraper,
 	ctx context.Context,
 	client graphql.Client,
-	repos []SearchNode,
+	repoName string,
+	owner string,
+	now pcommon.Timestamp,
+) (int, error) {
+	prOpenCount, err := ghs.getPrCount(ctx, client, repoName, ghs.cfg.GitHubOrg, []PullRequestState{PullRequestStateOpen})
+	if err != nil {
+		return 0, err
+	}
+	ghs.logger.Sugar().Debugf("open pull request count: %v for repo %v", prOpenCount, repoName)
+	ghs.mb.RecordGitRepositoryPullRequestCountDataPoint(now, int64(prOpenCount), repoName)
+
+	prMergedCount, err := ghs.getPrCount(ctx, client, repoName, ghs.cfg.GitHubOrg, []PullRequestState{PullRequestStateMerged})
+	if err != nil {
+		return 0, err
+	}
+
+	totalPrCount := add(prOpenCount, prMergedCount)
+	prPages := getNumPages(float64(100), float64(totalPrCount))
+	ghs.logger.Sugar().Debugf("pull request pages: %v for repo %v", prPages, repoName)
+	return prPages, err
+}
+
+func getPrData(
+	ghs *githubScraper,
+	ctx context.Context,
+	client graphql.Client,
+	prPages int,
+	repoName string,
+	owner string,
+) ([]PullRequestNode, error) {
+	var prCursor *string
+	var pullRequests []PullRequestNode
+
+	for i := 0; i < prPages; i++ {
+		pr, err := getPullRequestData(ctx, client, repoName, ghs.cfg.GitHubOrg, 100, prCursor)
+		if err != nil {
+			return nil, err
+		}
+
+		pullRequests = append(pullRequests, pr.Repository.PullRequests.Nodes...)
+		prCursor = &pr.Repository.PullRequests.PageInfo.EndCursor
+
+		if !pr.Repository.PullRequests.PageInfo.HasNextPage {
+			break
+		}
+	}
+	return pullRequests, nil
+}
+func getPullRequests(
+	ghs *githubScraper,
+	ctx context.Context,
+	client graphql.Client,
+	repos []SearchNodeRepository,
 	now pcommon.Timestamp,
 	pullRequestCh chan []PullRequestNode,
```

Comment:
Does this pull request channel actually need to be passed in? Or can it locally be declared within the function, thereby making the call within the function the async call?